src/main/java/slimeknights/tconstruct/tools/modifiers/ability/PlaceBlockModifier.java

```diff
@@ -0,0 +1,249 @@
+package slimeknights.tconstruct.tools.modifiers.ability;
+
+import javax.annotation.Nullable;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.SoundType;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.BlockItemUseContext;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.ItemUseContext;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.state.Property;
+import net.minecraft.state.StateContainer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.SoundEvent;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.shapes.ISelectionContext;
+import net.minecraft.world.World;
+import slimeknights.tconstruct.library.Util;
+import slimeknights.tconstruct.library.modifiers.SingleUseModifier;
+import slimeknights.tconstruct.library.tools.nbt.IModifierToolStack;
+import slimeknights.tconstruct.library.tools.nbt.ToolStack;
+
+/** Modifier version of a {@link net.minecraft.item.BlockItem}
+ * */
+public abstract class PlaceBlockModifier extends SingleUseModifier /*, BlockItem*/ {

```

Comment:
I feel like this is way overcomplicated for something that would be as simple as an `onItemUse` hook. For the blocks that tools will actually place, we don't need 90% of this code. Notably, we have no need for a tile entity placing behavior right now, and I'd rather not have that much dead code sitting untested. Plus a lot of this is hardcoded in a way you cannot have multiple block placement modifiers.

Just add a hook into `onItemUse`, and place the block using `setBlockState`.